sunpy/map/mapbase.py

```diff
@@ -1694,16 +1694,15 @@ def plot(self, annotate=True, axes=None, title=True, **imshow_kwargs):
             axes = wcsaxes_compat.gca_wcs(self.wcs)
 
         if not _basic_plot:
-            # Check that the image is properly oriented
-            if (not wcsaxes_compat.is_wcsaxes(axes) and
-                    not np.array_equal(self.rotation_matrix, np.identity(2))):
-                warnings.warn("This map is not properly oriented. Plot axes may be incorrect.",
-                              SunpyUserWarning)
-
-            elif not wcsaxes_compat.is_wcsaxes(axes):
+            if not wcsaxes_compat.is_wcsaxes(axes):
                 warnings.warn("WCSAxes not being used as the axes object for this plot."
-                              " Plots may have unexpected behaviour.",
+                              " Plots may have unexpected behaviour. To fix this pass "
+                              "'projection=map' when creating the axes",
                               SunpyUserWarning)
+                # Check if the image is properly oriented
+                if not np.array_equal(self.rotation_matrix, np.identity(2)):
+                    warnings.warn("This map is not properly oriented. Plot axes may be incorrect.",
```

Comment:
I think this message could be better, and we should take this opportunity to fix it. How's this for a suggestion:
```suggestion
                    warnings.warn("The axes of this map are not aligned to the pixel grid. Plot axes may be incorrect.",
```